File: code-coaches/medium/secret-message/secret_message.rb
def reverse_char(char)
  if char.match?(/[a-zA-Z]/)
    ascii_lowercase_a = 97
    ascii_lowercase_z = 122
    return (ascii_lowercase_z - char.downcase.ord + ascii_lowercase_a).chr
  end

  char
end

def encode_message(message)
  message.chars.map { |char| reverse_char(char) }.join
end

def main
  message = gets.chomp
  puts encode_message(message)
end

if __FILE__ == $0
  main
end

File: code-coaches/medium/secret-message/secret_message.py
def reverse_char(char: str) -> str:
    if char.isalpha():
        ascii_lowercase_a = 97
        ascii_lowercase_z = 122
        return chr(ascii_lowercase_z - ord(char.lower()) + ascii_lowercase_a)

    return char


def encode_message(message: str) -> str:
    return "".join(map(reverse_char, message))


def main():
    message = input()
    print(encode_message(message))


if __name__ == "__main__":
    main()
